src/XamlX/Ast/Clr.cs

```diff
@@ -640,7 +641,7 @@ public XamlILNodeEmitResult Emit(XamlEmitContext<IXamlILEmitter, XamlILNodeEmitR
         {
             var so = context.Configuration.WellKnownTypes.Object;
             var isp = context.Configuration.TypeMappings.ServiceProvider;
-            var subType = context.CreateSubType("XamlClosure_" + Guid.NewGuid(), so);
+            var subType = context.CreateSubType("XamlClosure_" + DeterministicName.GetString(), so);
```

Comment:
That is the ONLY relevant change in the entire PR. Unfortunately, your DeterministicName implementation is not deterministic.

Reply:
Brutal!